src/main/java/org/jabref/model/metadata/MetaData.java

```diff
@@ -64,9 +64,15 @@ public MetaData() {
         return Optional.ofNullable(saveOrderConfig);
     }
 
-    public void setSaveOrderConfig(SaveOrderConfig saveOrderConfig) {
+    public void setSaveOrderConfig(SaveOrderConfig saveOrderConfig, ChangePropagation postChanges) {
```

Comment:
You are adding extending the method signatures in this core class, and I do not like that. The `ChangePropagation` parameter is added to every method and that pollutes the signatures. 

Please remove `ChangePropagation` from all method signatures. Instead, add an additional constructor for `MetaData` that takes the parameter and store it in an instance variable. This should have the same effect, but cause much less code changes.

Reply:
That was also my first idea but then I saw [MetaData.java#L249](https://github.com/JabRef/jabref/blob/master/src/main/java/org/jabref/model/metadata/MetaData.java#L249) which was already there and also uses `ChangePropagation` in this context. However now we are using a `set` method to avoid the signature pollution :).